app/models/deploy.rb

```diff
@@ -130,19 +130,43 @@ def self.expired
   end
 
   def buddy_name
-    user.id == buddy_id ? "bypassed" : buddy.try(:name)
+    user.id == buddy_id ? "bypassed" : buddy_with_deleted.try(:name)
   end
 
   def buddy_email
-    user.id == buddy_id ? "bypassed" : buddy.try(:email)
+    user.id == buddy_id ? "bypassed" : buddy_with_deleted.try(:email)
   end
 
   def url
     AppRoutes.url_helpers.project_deploy_url(project, self)
   end
 
+  def to_csv
+    [
+      id, project_with_deleted.name, summary, commit, job.status, updated_at,
+      start_time, user_with_deleted.try(:name), user_with_deleted.try(:email), buddy_name, buddy_email,
+      stage_with_deleted.production, stage_with_deleted.no_code_deployed, project_with_deleted.deleted_at
+    ]
+  end
+
   private
 
+  def buddy_with_deleted
+    buddy || (buddy_id && User.with_deleted { User.find(buddy_id) })
+  end
+
+  def project_with_deleted
+    Project.unscoped { Project.find(stage_with_deleted.project_id) }
+  end
+
+  def stage_with_deleted
+    Stage.unscoped { Stage.find(stage_id) }
+  end
+
+  def user_with_deleted
+    User.unscoped { job.user_id > 0 ? User.find(job.user_id) : nil }
+  end
+
```

Comment:
test on in-depth and the rest can be copy-paste ?

On Thu, Jul 14, 2016 at 9:47 AM, Robert Ikeoka <notifications@github.com>
wrote:

> In app/models/deploy.rb
> <https://github.com/zendesk/samson/pull/1131#discussion_r70840556>:
>
> > +  def buddy_with_deleted
> > +    buddy || (buddy_id && User.with_deleted { User.find(buddy_id) })
> > +  end
> > +
> > +  def project_with_deleted
> > +    Project.unscoped { Project.find(stage_with_deleted.project_id) }
> > +  end
> > +
> > +  def stage_with_deleted
> > +    Stage.unscoped { Stage.find(stage_id) }
> > +  end
> > +
> > +  def user_with_deleted
> > +    User.unscoped { job.user_id > 0 ? User.find(job.user_id) : nil }
> > +  end
> > +
>
> Testing each of these associations is needed? I think it's a bit overkill
> to do that.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/zendesk/samson/pull/1131/files/be649820fdf2ba2f677a17a345a92967c5b8f28d#r70840556>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAAsZ4q-fdKkOugNptHfcAHTlGntBkK9ks5qVmgpgaJpZM4JIewY>
> .
>